src/api/ApiResponder.php

```diff
@@ -783,6 +783,125 @@ protected function get_interface_response_dismissGame($interface, $args) {
+     * @param array $args
+     * @return NULL|array
+     */
+    protected function get_interface_response_createTournament($interface, $args) {
+        if (!(array_key_exists('description', $args))) {
+            $args['description'] = '';
+        }
+
+        return $interface->tournament()->create_tournament(
+            $this->session_user_id(),
+            $args['tournamentType'],
+            $args['nPlayer'],
+            $args['maxWins'],
+            $args['description']
+        );
+    }
+
+    /**
+     * Interface redirect for loadTournamentData
+     *
+     * @param BMInterface $interface
+     * @param array $args
+     * @return NULL|array
+     */
+    protected function get_interface_response_loadTournamentData($interface, $args) {
+        return $interface->load_api_tournament_data(
+            $this->session_user_id(),
+            $args['tournament']
+        );
+    }
+
+    /**
+     * Interface redirect for updateTournament
+     *
+     * @param BMInterface $interface
+     * @param array $args
+     * @return NULL|array
+     */
+    protected function get_interface_response_updateTournament($interface, $args) {
+        if (array_key_exists('buttonNames', $args)) {
+            $button_name_array = $args['buttonNames'];
+        } else {
+            $button_name_array = NULL;
+        }
+
+        return $interface->tournament()->act_on_tournament(
+            $this->session_user_id(),
+            $args['tournamentId'],
+            $args['action'],
+            $button_name_array
+        );
+    }
+
+    /**
+     * Interface redirect for dismissTournament
+     *
+     * @param BMInterface $interface
+     * @param array $args
+     * @return NULL|bool
+     */
+    protected function get_interface_response_dismissTournament($interface, $args) {
+        $retval = $interface->tournament()->dismiss_tournament($this->session_user_id(), $args['tournamentId']);
+        if (isset($retval)) {
+            $interface->player()->update_last_action_time($this->session_user_id());
+        }
+        return $retval;
+    }
+
+    /**
+     * Interface redirect for followTournament
+     *
+     * @param BMInterface $interface
+     * @param array $args
+     * @return NULL|bool
+     */
+    protected function get_interface_response_followTournament($interface, $args) {
+        $retval = $interface->tournament()->watch_tournament($this->session_user_id(), $args['tournamentId']);
+        if (isset($retval)) {
+            $interface->player()->update_last_action_time($this->session_user_id());
+        }
+        return $retval;
+    }
+
+    /**
+     * Interface redirect for unfollowTournament
+     *
+     * @param BMInterface $interface
+     * @param array $args
+     * @return NULL|bool
+     */
+    protected function get_interface_response_unfollowTournament($interface, $args) {
+        $retval = $interface->tournament()->unwatch_tournament($this->session_user_id(), $args['tournamentId']);
```

Comment:
Reminder since you probably haven't been in this code for awhile: in order to make `unfollow` be the interface for non-participants only, you'll have to create a new `unfollow_tournament()` function that calls `unwatch_tournament()` only if conditions are met.  (This is the way `dismiss_tournament()` works now, and you'll have to do it precisely because `dismiss_tournament()` calls `unwatch_tournament()` via its own code path with different preconditions, as discussed in the state diagram conversation.